src/client/cl_main.cpp

```diff
@@ -4043,3 +4046,128 @@ void CL_GetVMGLConfig(vmglconfig_t *vmglconfig) {
 	vmglconfig->stereoEnabled = cls.glconfig.stereoEnabled;
 	vmglconfig->smpActive = cls.glconfig.smpActive;
 }
+
+void CL_ShaderStateChanged( void ) {
+	// Originally copied from CG_ShaderStateChanged, but rewritten to avoid issues of the original implementation and to
+	// support settings
+	char originalShader[MAX_QPATH];
+	char newShader[MAX_QPATH];
+	char settings[32];
+	char *timeOffset;
+	char *lightmapMode;
+	char *styleMode;
+
+	shaderRemapLightmapType_t lightmapModeValue;
+	shaderRemapStyleType_t styleModeValue;
+
+	const char *curPos = cl.gameState.stringData + cl.gameState.stringOffsets[ CS_MV_REMAPS ];
+	const char *endPos = curPos + strlen( curPos );
+
+	// Check configstring for prefix to avoid misinterpreting configstrings from mods
+	if ( !curPos || !*curPos || strncmp(curPos, "mvremap:", 8) ) return;
+
+	// Handling of these is really ugly. I originally wanted to handle them like the base game/cgame modules do, but
+	// those are prone to injections. Summary of delimeter issues:
+	//  - base uses '=', ':' an '@', all of them can be used in shader names
+	//  - .shader files don't seem to support spaces in shader names, but texture paths support spaces
+	//  - .shader files allow most other characters, including backspace '\'
+	//  -> using a delimeter is not a reliable option
+
+	// The next best thing seemed to be to give the length of the shader before its name to allow all characters to be
+	// used. As I don't want to waste any space (configstrings containing shader paths are wasteful as it is), I decided
+	// to encode the length in a single byte (MAX_QPATH is 80, so a single byte can easily hold the length). However the
+	// netcode and the handling of quotes lead to other issues:
+	//  - 34 maps to 32, thus 32 could mean either of these numbers while 34 doesn't exist
+	//  - 37 and >127 map to 46, thus 37 doesn't exist and 46 could mean 37, 46 or >127
+	//  -> the biggest amount of consecutive numbers without multiple meanings spans from 47 to 127; as MAX_QPATH is 80
+	//     this should fix exactly
+	//  -> could also use the range from 38 to 127, because - when isolating the range - there are no duplicates within
+	//     it; this would allow for a size of up to 89, which is unlikely to be useful if MAX_QPATH ever gets increased
+	//  -> using 47 to 127
+
+	// Skip prefix
+	curPos += 8;
+
+	int length;
+
+	// Parse configstring
+	while ( curPos && *curPos )
```

Comment:
You can get buffer overflow here in *curPos statement when coming from previous iteration of the loop. Maybe replace it with curPos < endPos instead?